src/System.Management.Automation/engine/Interop/Windows/WNetAddConnection2.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Windows
+    {
+        internal const int CONNECT_NOPERSIST = 0x00000000;
+        internal const int CONNECT_UPDATE_PROFILE = 0x00000001;
+        internal const int RESOURCE_GLOBALNET = 0x00000002;
+        internal const int RESOURCETYPE_ANY = 0x00000000;
+        internal const int RESOURCEDISPLAYTYPE_GENERIC = 0x00000000;
+        internal const int RESOURCEUSAGE_CONNECTABLE = 0x00000001;
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct NetResource
+        {
+            public int Scope;
+            public int Type;
+            public int DisplayType;
+            public int Usage;
+            public char* LocalName;
+            public char* RemoteName;
+            public char* Comment;
+            public char* Provider;
+        }
+
+        [LibraryImport("mpr.dll", EntryPoint = "WNetAddConnection2W", StringMarshalling = StringMarshalling.Utf16)]
+        internal static partial int WNetAddConnection2(ref NetResource netResource, byte[] password, string userName, int flags);
```

Comment:
Is the standard to keep these as the raw integer type or actually define an enum/flags enum? I usually do the latter for these types of fields as I find it easier to understand what "flags" can be set but curious if the dotnet guidelines state otherwise

Reply:
We could create the enum but this will not make the main code more readable rather the opposite - the name in callsite will be even longer. I think it is enough that these constants are already packed into a separate class.

Reply:
I can't say I agree with that, you are still going to do `Windows.INT_CONST1 | Windows.INT_CONST2`, with the enum you are now scoping it to specific values that the caller knows is valid here. This question was mostly for my curiosity's sake so if this is the preferred way then I'm fine with it.

Reply:
It's a compromise. If it was already in the .Net repo, I would just copy it. There's still a lot of work to be done for migration to LibraryImport and I'd like to get it done quickly while we're at the beginning of the milestone so we have time to stabilize. Then we could take our time to "standardize" this code.

Reply:
That's fair enough, I don't want to hold up this PR I'm just curious about the different standards people have to see if I can improve how I do things myself.